src/ModelContextProtocol/Client/AutoDetectingClientSessionTransport.cs

```diff
@@ -0,0 +1,196 @@
+            await _sseTransport.SendMessageAsync(message, cancellationToken).ConfigureAwait(false);
+            
+            // Signal that we have established a connection
+            LogUsingSSE(_name);
+            _delegatingChannelReader.SetConnected();
+        }
+        catch (Exception ex)
+        {
+            LogSSEConnectionFailed(_name, ex);
+            capturedEx = ex;
+            
+            try
+            {
+                if (_sseTransport != null)
+                {
+                    await _sseTransport.DisposeAsync().ConfigureAwait(false);
+                }
+            }
+            finally
+            {
+                // Set the error so the channel reader will propagate it
+                _delegatingChannelReader.SetError(ex);
+            }
+            
+            throw;
+        }
+    }
+
+    public async ValueTask DisposeAsync()
+    {
+        try
+        {
+            if (_streamableHttpTransport != null)
+            {
+                await _streamableHttpTransport.DisposeAsync().ConfigureAwait(false);
+                _streamableHttpTransport = null;
+            }
+            
+            if (_sseTransport != null)
+            {
+                await _sseTransport.DisposeAsync().ConfigureAwait(false);
+                _sseTransport = null;
+            }
+        }
+        catch (Exception ex)
+        {
+            LogDisposeFailed(_name, ex);
+        }
+    }
+    
+    [LoggerMessage(Level = LogLevel.Debug, Message = "{EndpointName}: Attempting to connect using Streamable HTTP transport")]
+    private partial void LogAttemptingStreamableHttp(string endpointName);
+    
+    [LoggerMessage(Level = LogLevel.Debug, Message = "{EndpointName}: Streamable HTTP transport failed with status code {StatusCode}, falling back to SSE transport")]
+    private partial void LogStreamableHttpFailed(string endpointName, System.Net.HttpStatusCode statusCode);
+    
+    [LoggerMessage(Level = LogLevel.Debug, Message = "{EndpointName}: Streamable HTTP transport failed with exception, falling back to SSE transport")]
+    private partial void LogStreamableHttpException(string endpointName, Exception exception);
+    
+    [LoggerMessage(Level = LogLevel.Debug, Message = "{EndpointName}: Using Streamable HTTP transport")]
+    private partial void LogUsingStreamableHttp(string endpointName);
+    
+    [LoggerMessage(Level = LogLevel.Debug, Message = "{EndpointName}: Attempting to connect using SSE transport")]
+    private partial void LogAttemptingSSE(string endpointName);
+    
+    [LoggerMessage(Level = LogLevel.Debug, Message = "{EndpointName}: Using SSE transport")]
+    private partial void LogUsingSSE(string endpointName);
+    
+    [LoggerMessage(Level = LogLevel.Error, Message = "{EndpointName}: SSE transport connection failed")]
+    private partial void LogSSEConnectionFailed(string endpointName, Exception exception);
```

Comment:
```suggestion
    [LoggerMessage(Level = LogLevel.Error, Message = "{EndpointName}: Failed to connect using both Streamable HTTP and SSE transports.")]
    private partial void LogAutoDetectingConnectionFailed(string endpointName, Exception exception);
```

Also, add periods to the end of all the log messages.